libs/server/repository/core/application-services/src/lib/commands/add-repository/add-repository.handler.ts

```diff
@@ -7,9 +7,16 @@ export class AddRepositoryHandler implements ICommandHandler<AddRepositoryComman
   constructor(private repositoryRepository: RepositoryRepository) {}
 
   async execute(command: AddRepositoryCommand): Promise<void> {
-    const repository = await this.repositoryRepository.getSingleRepositoryByName(
-      command.repositoryName
-    );
+    const { repositoryName, maxLines, maxWaitingTime } = command;
+
+    let repository = await this.repositoryRepository.getSingleRepositoryByName(repositoryName);
+
+    repository = {
```

Comment:
repository has to be a Repository class instance. We can't use spread operator because it will loose a type. 
Do just repository.maxLines = maxLines
...

Reply:
eventually, you can use plainToClass after the spread